sunpy/net/dataretriever/client.py

```diff
@@ -259,7 +261,13 @@ def _get_time_for_url(self, urls):
 
         It should return a sunpy.time.TimeRange object per URL.
         """
-        return NotImplemented
+        if self.scraper is None:
+            return NotImplemented
+        else:
+            scraper = self.scraper
+            almost_day = TimeDelta(1 * u.day - 1 * u.millisecond)
+            times = [TimeRange(t0, t0 + almost_day) for t0 in map(scraper._extractDateURL, urls)]
```

Comment:
Why is this assuming one day? The base class should not be making assumptions about the form of the results returned by the clients.

Reply:
Yes, exactly my doubts were regarding this. What does the `start time` and `end time` in the response table represent? Since the data files we get through any query are always associated with timestamps, and if there are multiple files on the same date, what should set the start and end time to? Should they be timeranges of one day, or shoud timeranges should be divided ? (say there are 10 files for one day, so one day is divided in 10 timeranges?)

Reply:
Yes, I don't think it should assume one day. I think the proper way to fix this is to modify the individual `_extractDateURL` methods to return an interval (ie. start time and end time) instead of currently just returning a start time. That way the length of an interval can be specified on a per-client basis.